hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java

```diff
@@ -107,11 +105,8 @@ private ClassPool buildClassPool(EnhancementContext enhancementContext) {
 	}
 
 	private CtClass loadCtClassFromClass(ClassPool cp, Class<?> aClass) throws IOException {
-		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( getFilenameForClass( aClass ) ) );
-	}
-
-	private String getFilenameForClass(Class<?> aClass) {
-		return aClass.getName().replace( '.', File.separatorChar ) + JavaFileObject.Kind.CLASS.extension;
+		String resourceName = aClass.getName().replace( '.', '/' ) + ".class";
+		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( resourceName ) );
```

Comment:
I'm going to fix it

On 10 July 2015 at 20:20, Scott Marlow <notifications@github.com> wrote:

> In
> hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java
> <https://github.com/hibernate/hibernate-orm/pull/996#discussion_r34387439>
> :
>
> > @@ -107,11 +105,8 @@ private ClassPool buildClassPool(EnhancementContext enhancementContext) {
> >  	}
> >
> >  	private CtClass loadCtClassFromClass(ClassPool cp, Class<?> aClass) throws IOException {
> > -		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( getFilenameForClass( aClass ) ) );
> > -	}
> > -
> > -	private String getFilenameForClass(Class<?> aClass) {
> > -		return aClass.getName().replace( '.', File.separatorChar ) + JavaFileObject.Kind.CLASS.extension;
> > +		String resourceName = aClass.getName().replace( '.', '/' ) + ".class";
> > +		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( resourceName ) );
>
> I wonder what closes the stream returned by getResourceAsStream()? Should
> we close that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hibernate/hibernate-orm/pull/996/files#r34387439>.
>